internal/storage/image.go

```diff
@@ -298,6 +306,30 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 	return results, nil
 }
 
+// determines if an images is intermediate
+func (svc *imageService) isIntermediate(image *storage.Image) (bool, error) {
+	storageLayers, err := svc.GetStore().Layers()
```

Comment:
nit: I feel like we're needlessly repeating a call to GetStore().Layers(). We could save the layers above when we do the images, and pass the layers in with the image to `isIntermediate` (and make isIntermediate not rely on svc). this would prevent us calling Layers() for each image (which could be an expensive operation, idk)

Reply:
fixed !